src/system_identity/identity_creator.rs

```diff
@@ -0,0 +1,50 @@
+use super::creation_response::SystemIdentityCreationResponseData;
+
+/// Interface describing being able to create L2 System Identities.
+pub trait L1IdentityCreator {
+    // TODO type Output;
+    type Error: std::error::Error;
+    fn create_l1_system_identity(&self) -> Result<SystemIdentityCreationResponseData, Self::Error>;
+}
+
+/// Interface describing being able to create L2 System Identities.
+pub trait L2IdentityCreator {
+    // TODO type Output;
+    type Error: std::error::Error;
+    fn create_l2_system_identity(
+        &self,
+        pub_key: &[u8],
```

Comment:
it is a bit confusing that just pub_key is a parameter, since there is also the `name` parameter needed , or perhaps others i'm missing . Would make sense to add those  or remove all and expect the implementation knows them (defined at built time)

Reply:
This was a suggestion by @sigilioso given the pub key is generated elsewhere and passed to it, as in https://github.com/newrelic/newrelic-auth-rs/blob/ed86a2b55ba27783b55ed7f4aa8d043c8f1fad93/src/system_identity/generator.rs#L28-L37